torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -296,13 +296,28 @@ void FoldPrePackingOps(script::Module& m) {
   PrePackingOpsFolder(m, filter_fn, "prepack_folding");
 }
 
-c10::optional<script::Module> optimizeForMobile(const script::Module& m) {
+script::Module optimizeForMobile(
+    const script::Module& m,
+    const std::map<MobileOptimizerType, bool>& whitelist_optimizers) {
   auto cloned_module = m.clone();
   cloned_module.eval();
-  cloned_module = FoldConvBatchNorm2d(cloned_module);
-  insertPrePackedOps(cloned_module);
-  cloned_module = freeze_module(cloned_module);
-  FoldPrePackingOps(cloned_module);
+
+  auto bn_it =
```

Comment:
I would personally mark these as const auto (constness is not implied otherwise here) to have the compiler help me in situations where I accidentally modify the variable, like for instance `if (bn_it = whitelist_optimizers.end())`.